packages/fast-form-generator-react/src/form/form-item.array.tsx

```diff
@@ -98,10 +121,38 @@ class FormItemArray extends React.Component<IFormItemArrayProps & IManagedClasse
         );
     }
 
+    public componentDidUpdate(): void {
+        this.state.hideOptionMenu
+            ? window.removeEventListener("click", this.handleWindowClick)
+            : window.addEventListener("click", this.handleWindowClick);
```

Comment:
Every update will perform this, even if it's not needed. Do we need to do this on update every time the cycle is ran? Is there a reason we should not simply add the event listener on mount and remove on unmount?